crates/red_knot_python_semantic/src/semantic_index/constraint.rs

```diff
@@ -7,13 +7,19 @@ use crate::semantic_index::expression::Expression;
 use crate::semantic_index::symbol::{FileScopeId, ScopeId};
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
-pub(crate) enum Constraint<'db> {
+pub(crate) struct Predicate<'db> {
+    pub(crate) node: PredicateNode<'db>,
+    pub(crate) negative: bool,
```

Comment:
It feels slightly odd to me that a negative boolean value for this field indicates a positive predicate. Could we have the meaning flipped, so that a positive value for the field indicates a positive predicate (and vice versa)?

```suggestion
    pub(crate) is_positive: bool,
```